Implement share password settings

st3iny opened the pull request:
Fix #2644 

## How to test?
1. Configure your dev server to be able to send emails.
2. Create a circle.
3. Add an external member to the circle by email (you have to be able to receive emails for this address).
4. Test random password:
4.1. Enforce share passwords for the circle (but not a unique password).
4.2. Share a file with the circle.
4.3. Observe that the external user gets 2 mails (one with the share and one with a random password).
5. Test unique password:
5.1. Configure a unique password (second checkbox + input).
5.2. Share another file with the circle.
5.3. Observe that the external user gets a singe mail containing the share.
5.4. Unlock the share using the unique password from 5.1.

## Case 1: Configuring a new unique password

![new-password](https://user-images.githubusercontent.com/1479486/173560989-00d88d9a-e2cd-44af-8f53-fedec5177519.gif)


## Case 2: Changing the unique password when one is already set

I added a button for this case because the password is hashed and stored. It is impossible to reverse this process so it does not make sense to show a prefilled input field in this case.

![change-password](https://user-images.githubusercontent.com/1479486/173561012-fef78df4-432b-4ac5-9185-e9c96c5816f2.gif)

## Case 3: An error happens while saving the password

![change-password-error](https://user-images.githubusercontent.com/1479486/173561084-cf565fad-9b3e-491c-96e2-4542d39ebc84.gif)

## Case 4: Disabling password protection

![remove-password](https://user-images.githubusercontent.com/1479486/173561127-24de10c2-a662-484b-850b-203f2b94229a.gif)


## TODO
- [x] Enforce share password checkbox
- [x] Use unique password checkbox
- [x] Unique password input field
- [x] Test if feature/backend is working

ArtificialOwl commented:
Excellent !

From my point of view, there is still some margin for a small improvement regarding the creation of a single password:
Instead of saving the password like right now, add a button  '[Store]' or '[Save]', at the right of the `field`. When clicked, and only when clicked, the password is stored and the `field` is replaced by the button `[Change unique password]` to be in the same situation than the page is refreshed.

Maybe a `[Cancel]` button when displaying the `field`+`[Store]` (after clicking on `[Change unique password]`) is overkill ?
The idea is that, if the page is opened, an admin does not reset the password by mistake.

st3iny commented:
> Excellent !
> 
> From my point of view, there is still some margin for a small improvement regarding the creation of a single password: Instead of saving the password like right now, add a button '[Store]' or '[Save]', at the right of the `field`. When clicked, and only when clicked, the password is stored and the `field` is replaced by the button `[Change unique password]` to be in the same situation than the page is refreshed.

I added a save button and removed the indicator.

> Maybe a `[Cancel]` button when displaying the `field`+`[Store]` (after clicking on `[Change unique password]`) is overkill ? The idea is that, if the page is opened, an admin does not reset the password by mistake.

I think that a dedicated save button suffices. The new password will be saved only if the save button is clicked. If an admin opens the page, clicks change password and reloads the page, the old password will still be intact.

---

I also updated all gifs above.

st3iny commented:
Rebased and squashed.
